ruma-api/src/lib.rs

```diff
@@ -254,6 +255,59 @@ pub trait OutgoingRequest {
     ) -> Result<http::Request<Vec<u8>>, IntoHttpError>;
 }
 
+/// An extension to `OutgoingRequest` which provides Appservice specific methods
+pub trait OutgoingRequestAppserviceExt: OutgoingRequest + Sized {
+    /// Tries to convert this request into an `http::Request` and appends a virtual `user_id` to
+    /// [assert Appservice identity].
+    ///
+    /// [assert Appservice identity]: https://matrix.org/docs/spec/application_service/r0.1.2#identity-assertion
```

Comment:
```suggestion
    /// [assert Appservice identity][id_assert].
    ///
    /// [id_assert]: https://matrix.org/docs/spec/application_service/r0.1.2#identity-assertion
```

Then you don't go over 100 chars line limit.